### ISSUE TITLE
Doubts about reward

### ISSUE THREAD

**Issue by Mryangkaitong** [lvwerra/trl#173]
Thank you very much for such a great work. When I run gpt2-sentiment.py (https://github.com/lvwerra/trl/blob/main/examples/sentiment/scripts/gpt2-sentiment.py#L151), I have a question I would like to ask:
`rewards = [torch.tensor(output[1]["score"]) for output in pipe_outputs] `
What is the value range of the reward score here(I print to see that there are greater than 1, greater than 2, etc.), is it directly taken logits? And rewards can be any range? 
For each input sentence, how is his reward obtained? Is it the sum of the logits of each token in the sentence?

**Comment by lvwerra** [comment]
Yes, we use the raw logits, we found that usually works better than e.g. softmax normalized outputs. And you get one reward per sequence from the classifier model so no need to aggregate them. In theory you could pass a reward per token but this is not implemented at the moment.

**Comment by Mryangkaitong** [comment]
Thank you very much, I have another question, that is, the model I want to train with ppo is AutoModelForSeq2SeqLM, just refer to here(https://github.com/lvwerra/trl/blob/main/examples/sentiment/scripts/gpt2-sentiment.py#L104) and change it to
`from transformers import  AutoModelForSeq2SeqLM
model = AutoModelForSeq2SeqLM.from_pretrained(config.model_name, trust_remote_code=True)
ref_model = AutoModelForSeq2SeqLM.from_pretrained(config.model_name, trust_remote_code=True)`

But get the following error:
`alueError: model must be a PreTrainedModelWrapper, got <class 'transformers_modules.local.modeling_glm.GLMForConditionalGeneration'> - supported architectures are: (<class 'trl.models.modeling_value_head.AutoModelForCausalLMWithValueHead'>, <class 'trl.models.modeling_value_head.AutoModelForSeq2SeqLMWithValueHead'>)`

How can I adapt in the current trl framework so that I can use ppo to train my AutoModelForSeq2SeqLM model? 
Thank you very much, looking forward to answer

**Comment by lvwerra** [comment]
@younesbelkada this might be an issue of a kwarg not being passed along to the original model class, wdyt?

**Comment by younesbelkada** [comment]
Hello @Mryangkaitong, 
Thanks for the issue!
Yes this is possible, however the script below is working on the `main` branch of `trl`:
```python
from trl import AutoModelForSeq2SeqLMWithValueHead
model = AutoModelForSeq2SeqLMWithValueHead.from_pretrained("t5-small", trust_remote_code=True)
```
I believe you need to use the latest changes of the library, by installing `trl` from source:
```
pip install git+https://github.com/lvwerra/trl.git
```
Let us know if the problem still persists!

**Comment by Mryangkaitong** [comment]
ok , I have solved it
